Extension/src/LanguageServer/configurations.ts

```diff
@@ -241,23 +241,12 @@ export class CppProperties {
         });
 
         this.configFileWatcher.onDidChange(() => {
-            // If the file is one of the textDocument's vscode is tracking, we need to wait for an
-            // onDidChangeTextDocument event, or we may get old/cached contents when we open it.
-            let alreadyTracking: boolean = false;
-            for (let i: number = 0; i < vscode.workspace.textDocuments.length; i++) {
-                if (vscode.workspace.textDocuments[i].uri.fsPath === settingsPath) {
-                    alreadyTracking = true;
-                    break;
-                }
-            }
-            if (!alreadyTracking) {
-                this.handleConfigurationChange();
-            }
+            this.handleConfigurationChange();
         });
 
-        vscode.workspace.onDidChangeTextDocument((e: vscode.TextDocumentChangeEvent) => {
+        vscode.workspace.onDidChangeTextDocument((e) => {
             if (e.document.uri.fsPath === settingsPath) {
-                this.handleConfigurationChange();
+                this.handleSquiggles();
```

Comment:
Could you add the other change we discussed, to call `handleSquiggles()` as the file becomes visible?  And, could you avoid calling `handleSquiggles()` here if this document is not within the list of `vscode.window.visibleTextEditors`?

Reply:
Don't use onDidChangeActiveTextEditor.  There can only be one active editor, but there can be multiple that are visible.  You'll want to use onDidChangeVisibleTextEditors (which returns a list of the currently visible documents), keep track of whether the settings file is visible, and call handleSquiggles when it transitions from not-visible to visible.

Reply:
right I totally forgot about the need to iterate through the editors

Reply:
Also, you could you add a check here to onDidChangeTextDocument, to avoid calling handleSquiggles if the document is not visible?

Reply:
isn't this redundant? You wouldn't be able to change the json file unless it was visible

Reply:
@browntarik The user might use the JSON UI editor?

Reply:
Right, the user might edit the config in the config UI, or sync down an updated c_cpp_properties.json via a git pull.  The idea I'm advocating is to simply (and only) update squiggles when the file is modified AND visible, or when it becomes visible.